transform/apply.go

```diff
@@ -102,6 +119,9 @@ func Apply(
 	// until doneCh gets signaled.
 	go func() {
 		if !wait {
+			// if we're running this script async, bind to the background context
+			// note that we lose any values attached to the given context
+			ctx = svc.bgCtx
```

Comment:
Hmmm, I had a hunc these things would start cropping up with all the context passing. 
For this specific issue, I think it is warranted to actually create a `context.Background()` and re-attach values onto it. Mainly because sharing a global context in this sense will prevent us from multiplexing the values for different simultaneous requests/users without some gross name spacing or introducing a global lock which kills half the multi tenancy benefits.

Reply:
yeah I'm thinking a very similar thing, but instead of using `context.Background`, use the `bgCtx` here as the root. That way we get to a place we want: a context that will be cancelled when the process tries to exit gracefully (by closing the root) that also has request-scoped values on the child context. Pseudocode:

```go
// package identity
const identityKeyIDKey = "keyID"

func NewUserContext(ctx, newRoot context.Context) (context.Context, context.CancelFunc) {
  // only keyID value added for now, but this func lets us add more later if needed
  key, _ := keyFromContext(ctx)
  return newRoot.WithValue(identityKeyIDKey, key)
}

// at this call site:
go func() {
  if !wait {
    ctx = identity.NewUserContext(ctx, svc.bgCtx)
    // ...
```

Reply:
Yes, forgot that this is still request based and the parent has a scope as well.